### ISSUE TITLE
Parser doesn't resolve object to MapSchema

### ISSUE THREAD

**Issue by ackintosh** [swagger-api/swagger-parser#1071]
From my understanding a `object` which has `additionalProperties` should be resolved to MapSchema but it doesn't at the moment.

#### YAML

```yaml
...
...
...
paths:
  '/mapschema':
    get:
      description: MapSchema
      operationId: MapSchema
      responses:
        '200':
          description: Successful response.
          content:
            application/json:
              schema:
                type: object
                additionalProperties:
                  type: string
```

#### The schema is resolved as ObjectSchema instead of MapSchema

```java
OpenAPI result = (new OpenAPIV3Parser()).read("mapschema.yaml");

result.getPaths()
.get("/mapschema")
.getGet()
.getResponses()
.get("200")
.getContent()
.get("application/json")
.getSchema().getClass().getName()
// io.swagger.v3.oas.models.media.ObjectSchema
```

If my understanding is correct, I'd file a PR to fix it. 😉

**Comment by gracekarina** [comment]
Hi @ackintosh thanks for the report, we have merged a PR, that solved the issue, please let us know if this issue keeps happening.

**Comment by ackintosh** [comment]
@gracekarina Thanks for the fix. 👍

**Comment by frantuma** [comment]
Fixed by #1077 and #1082
